pkg/builddef/healthcheck.go

```diff
@@ -0,0 +1,186 @@
+package builddef
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/NiR-/zbuild/pkg/image"
+	"github.com/mitchellh/mapstructure"
+)
+
+// HealthcheckConfig represents healthcheck as specified in specialized
+// definition files.
+type HealthcheckConfig struct {
+	*HealthcheckHTTP `mapstructure:"http"`
+	*HealthcheckFCGI `mapstructure:"fcgi"`
+	*HealthcheckCmd  `mapstructure:"cmd"`
+
+	Type     HealthcheckType
+	Interval time.Duration
+	Timeout  time.Duration
+	Retries  int
+}
+
+func (hc *HealthcheckConfig) IsValid(allowedTypes []string) bool {
+	if hc == nil {
+		return true
```

Comment:
does this cover the case when no HC config is set ?

Reply:
When it's not set in zbuildfiles, it takes the default value defined by each kind, so it should never be nil. However as this is a method on a pointer value, I prefer to check that the pointer is not nil, otherwise the method might panic :wink: